src/back/msl/writer.rs

```diff
@@ -1713,7 +1715,17 @@ impl<W: Write> Writer<W> {
                     Mf::Reflect => "reflect",
                     Mf::Refract => "refract",
                     // computational
-                    Mf::Sign => "sign",
+                    Mf::Sign => match arg_type {
```

Comment:
You should use `arg_type.scalar_kind()` here.

Reply:
Thanks! Done in d16e605aa6f58043cef24b6cb8204bcf0dfb3cab